manifests/install.pp

```diff
@@ -18,7 +18,9 @@
     }
   }
 
-  Package { provider => $nodejs::package_provider }
+  if $nodejs::package_provider != undef {
```

Comment:
Technically this is not needed. When ```undef``` it'll fall back on the default value from puppet.